internal/status/status.go

```diff
@@ -31,10 +31,10 @@ import (
 	"errors"
 	"fmt"
 
-	"github.com/golang/protobuf/proto"
-	"github.com/golang/protobuf/ptypes"
 	spb "google.golang.org/genproto/googleapis/rpc/status"
 	"google.golang.org/grpc/codes"
+	"google.golang.org/protobuf/proto"
```

Comment:
I believe this causes an API change, right?  Now our status package only works with new messages, and breaks with the old ones?  IIUC we should be using this instead in `WithDetails` and `Details`: https://pkg.go.dev/google.golang.org/protobuf/protoadapt#MessageV1

Reply:
@Clement-Jean This changes the API of WithDetails, because `proto.Message` is now different.

Reply:
did you see the tests ([status_test.go](https://github.com/Clement-Jean/grpc-go/blob/remove_old_proto_pkg/internal/status/status_test.go))? You mean that this isn't enough to prove the API didn't change? If that's not enough maybe I can revert the changes or add more tests to make sure it doesn't break